Source/santad/Santad.mm

```diff
@@ -0,0 +1,281 @@
+                SNTExecutionController *exec_controller,
+                std::shared_ptr<SNTPrefixTree> prefix_tree) {
+  SNTConfigurator *configurator = [SNTConfigurator configurator];
+
+  SNTDaemonControlController *dc =
+    [[SNTDaemonControlController alloc] initWithAuthResultCache:auth_result_cache
+                                              notificationQueue:notifier_queue
+                                                     syncdQueue:syncd_queue
+                                                         logger:logger];
+
+  control_connection.exportedObject = dc;
+  [control_connection resume];
+
+  if ([configurator exportMetrics]) {
+    metrics->StartPoll();
+  }
+
+  SNTEndpointSecurityDeviceManager *device_client =
+    [[SNTEndpointSecurityDeviceManager alloc] initWithESAPI:esapi
+                                                     logger:logger
+                                            authResultCache:auth_result_cache];
+
+  device_client.blockUSBMount = [configurator blockUSBMount];
+  device_client.remountArgs = [configurator remountUSBMode];
+  device_client.deviceBlockCallback = ^(SNTDeviceEvent *event) {
+    [[notifier_queue.notifierConnection remoteObjectProxy]
+      postUSBBlockNotification:event
+             withCustomMessage:([configurator remountUSBMode]
+                                  ? [configurator bannedUSBBlockMessage]
+                                  : [configurator remountUSBBlockMessage])];
+  };
+
+  SNTEndpointSecurityRecorder *monitor_client =
+    [[SNTEndpointSecurityRecorder alloc] initWithESAPI:esapi
+                                                logger:logger
+                                              enricher:enricher
+                                    compilerController:compiler_controller
+                                       authResultCache:auth_result_cache
+                                            prefixTree:prefix_tree];
+
+  SNTEndpointSecurityAuthorizer *authorizer_client =
+    [[SNTEndpointSecurityAuthorizer alloc] initWithESAPI:esapi
+                                          execController:exec_controller
+                                      compilerController:compiler_controller
+                                         authResultCache:auth_result_cache];
+
+  SNTEndpointSecurityTamperResistance *tamper_client =
+    [[SNTEndpointSecurityTamperResistance alloc] initWithESAPI:esapi logger:logger];
+
+  EstablishSyncServiceConnection(syncd_queue);
+
+  NSArray<SNTKVOManager *> *kvoObservers = [[NSArray alloc]
+    initWithObjects:
+      [[SNTKVOManager alloc] initWithObject:configurator
+                                   selector:@selector(clientMode)
+                                       type:[NSNumber class]
+                                   callback:^(NSNumber *oldValue, NSNumber *newValue) {
+                                     if ([oldValue longLongValue] == [newValue longLongValue]) {
```

Comment:
Is this possible?

Reply:
Spoke offline. Don't think so in the normal case. There is an abnormal case where invalid types are configured which could result in the same values for old and new, but this would be an unknown state and doesn't change much. Removing.